cookbooks/aws-parallelcluster-environment/recipes/config/mount_home.rb

```diff
@@ -11,12 +11,115 @@
+end
+
+if shared_home == 'none'
+  # Mount the NFS export to compute and login nodes, the head node will export /home later
+  case node['cluster']['node_type']
+  when 'ComputeFleet', 'LoginNode'
+    volume "mount /home" do
+      action :mount
+      shared_dir '/home'
+      device(lazy { "#{node['cluster']['head_node_private_ip']}:#{node['cluster']['head_node_home_path']}" })
+      fstype 'nfs'
+      options node['cluster']['nfs']['hard_mount_options']
+      retries 10
+      retry_delay 6
+    end
+  when 'HeadNode'
+    Chef::Log.info("Do not mount NFS shares on the HeadNode")
+  else
+    raise "node_type must be ComputeFleet, LoginNode or HeadNode"
+  end
+else
+  # Identify the filesystem that is shared and mount it
+  include_recipe "aws-parallelcluster-environment::update_fs_mapping"
+  include_recipe "aws-parallelcluster-environment::backup_home_shared_data"
+  case shared_home
+  when 'efs'
+    shared_storage['efs'].each_with_index do |dir, index|
+      next unless dir == "/home" || dir == 'home'
+      efs "mount shared efs home" do
+        shared_dir_array [dir]
+        efs_fs_id_array [node['cluster']['efs_fs_ids'].split(',')[index]]
+        efs_encryption_in_transit_array [node['cluster']['efs_encryption_in_transits'].split(',')[index]]
+        efs_iam_authorization_array [node['cluster']['efs_iam_authorizations'].split(',')[index]]
+        action :mount
+      end
+      break
+    end
+  when 'fsx'
+    shared_storage['fsx'].each_with_index do |dir, index|
+      next unless dir == "/home" || dir == 'home'
+      lustre "mount shared fsx home" do
+        fsx_fs_id_array [node['cluster']['fsx_fs_ids'].split(',')[index]]
+        fsx_fs_type_array [node['cluster']['fsx_fs_types'].split(',')[index]]
+        fsx_shared_dir_array [dir]
+        fsx_dns_name_array [node['cluster']['fsx_dns_names'].split(',')[index]]
+        fsx_mount_name_array [node['cluster']['fsx_mount_names'].split(',')[index]]
+        fsx_volume_junction_path_array [node['cluster']['fsx_volume_junction_paths'].split(',')[index]]
+        action :mount
+      end
+      break
+    end
+  when 'ebs'
+    case node['cluster']['node_type']
+    when 'HeadNode'
+      shared_storage['ebs'].each_with_index do |dir, index|
+        next unless dir == "/home" || dir == 'home'
+        manage_ebs "add ebs /home" do
+          shared_dir_array [dir]
+          vol_array [node['cluster']['volume'].split(',')[index]]
+          action %i(mount export)
+        end
+        break
+      end
+    when 'ComputeFleet', 'LoginNode'
+      volume "mount /home" do
+        action :mount
+        shared_dir '/home'
+        device(lazy { "#{node['cluster']['head_node_private_ip']}:#{format_directory('/home')}" })
+        fstype 'nfs'
+        options node['cluster']['nfs']['hard_mount_options']
+        retries 10
+        retry_delay 6
+      end
+    end
+  when 'raid'
+    case node['cluster']['node_type']
+    when 'HeadNode'
+      raid "add raid /home" do
```

Comment:
In this case, I wanted to avoid having to check for both /home and home.  And this is guaranteed to be correct since we identified the directory prior.  The array conversion doesn't help here.  I did that so I could iterate over all of them in a single `each`